packages/form-js-playground/test/spec/Playground.spec.js

```diff
@@ -207,4 +299,143 @@ describe('playground', function() {
     expect(spy).to.have.been.called;
   });
 
+
+  describe('attach components', function() {
+
+    /**
+     * @typedef { {
+     *   name: String,
+     *   attachFn: String,
+     *   selector: String
+     * } } PlaygroundComponent
+     */
+
+    /** @type Array<PlaygroundComponent> */
+    const components = [
+      {
+        name: 'editor',
+        attachFn: 'attachEditorContainer',
+        selector: 'fjs-form-editor'
+      },
+      {
+        name: 'preview',
+        attachFn: 'attachPreviewContainer',
+        selector: 'fjs-form'
+      },
+      {
+        name: 'data',
+        attachFn: 'attachDataContainer',
+        selector: 'cm-editor'
+      },
+      {
+        name: 'result',
+        attachFn: 'attachResultContainer',
+        selector: 'cm-editor'
+      },
```

Comment:
Should these two have the same selector?

Reply:
I thought about adding a custom class name to each Code Mirror instance. However, it was not possible (at least not clearly documented). That's why I keep things as they are; you can still handle/style each editor via parent selectors.

Reply:
I was afraid that having the same selector in the tests can make us think data container is attached while it's the result container. If that's not the case, we can ignore it.

Reply:
As we now remove every container after each test, it shouldn't be the case.

Reply:
I think we are fine.